### PR TITLE
SER 2543: Make blank?, present?, and any? work for AF query objects

### PR DESCRIPTION
**Pull request by kencorley**

### CODE REVIEW

**Comment by kencorley** [comment]
[Make .blank? and .present? work in ActiveForce::ActiveQuery objects](https://trello.com/c/F2fIlCot/2543-make-blank-and-present-work-in-activeforceactivequery-objects)

File: spec/active_force/active_query_spec.rb [hunk 1]
```diff
@@ -40,6 +39,70 @@
...
+    context 'when there are no records' do
+      let(:api_result) { [] }
+
+      it 'returns true' do
+        result = active_query.where("Text_Label = 'foo'").blank?
+        expect(result).to be true
+      end
+
+      it 'returns true' do
+        result = active_query.where("Text_Label = 'foo'").first.blank?
```
Comment (sbelknap-bf) [inline spec/active_force/active_query_spec.rb:56]: this is not utilizing the delegation. this spec is basically confirming that you can call `.first` on an ActiveQuery object, find nil, and then call `.blank?` on that nil. you could comment out the delegator in the ActiveQuery model and this spec will still pass

File: spec/active_force/active_query_spec.rb [hunk 2]
```diff
@@ -40,6 +39,70 @@
...
+    end
+
+    context 'when records are returned' do
+      it 'returns false' do
+        result = active_query.where("Text_Label = 'foo'").blank?
+        expect(result).to be false
+      end
+
+      it 'returns false' do
+        result = active_query.where("Text_Label = 'foo'").first.blank?
```
Comment (sbelknap-bf) [inline spec/active_force/active_query_spec.rb:68]: similarly here, this is testing `.first` on the ActiveQuery object to return the first Object, and then calling `.blank?` on that object. not using the delegator

**Review by TABeauchat — APPROVED** [review]
Nice and simple addition!

**Comment by sbelknap-bf** [comment]
looks good, i would also consider adding `.any?` to the list of delegators... as that is the method that comes to mind for me when i want to check if there are any results to a query... whereas `.blank?` and `.present?` can be called on any Object, and may have previously been cause for confusion previously... because you could call `.present?` on an ActiveQuery Object  and get `true`, even if the ActiveQuery Object contained no records. Same thing for `.blank?`. But if I try calling `.any?` on an ActiveQuery Object, I will get a no method error

**Comment by kencorley** [comment]
> looks good, i would also consider adding `.any?` to the list of delegators... as that is the method that comes to mind for me when i want to check if there are any results to a query... whereas `.blank?` and `.present?` can be called on any Object, and may have previously been cause for confusion previously... because you could call `.present?` on an ActiveQuery Object and get `true`, even if the ActiveQuery Object contained no records. Same thing for `.blank?`. But if I try calling `.any?` on an ActiveQuery Object, I will get a no method error

Added.

**Review by sbelknap-bf — APPROVED** [review]
thanks for adding my `.any?` :) looks good

**Review by asedge — APPROVED** [review]
Nice work, thanks!

**Review by salonishah23 — COMMENTED** [review]
Good adds but it would be nice to understand where present? is used and whether that usage can be replaced with any? or exists? for an easy win

**Pull request merged**